sw/host/opentitanlib/src/io/jtag.rs

```diff
@@ -131,31 +131,225 @@ pub enum JtagTap {
     LcTap,
 }
 
-/// List of useful RISC-V general purpose registers
-#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
-pub enum RiscvGpr {
-    GP,
-    SP,
-    RA,
+macro_rules! def_enum_name {
+    ($(#[$attr:meta])* pub enum $name:ident {
+        $($reg:ident,)*
+    }) => {paste::paste!{
+        $(#[$attr])*
+        pub enum $name {
+            $($reg,)*
+        }
+
+        impl $name {
+            pub fn name(self) -> &'static str {
+                match self {
+                    $(Self::$reg => stringify!([<$reg:lower>]),)*
+                }
+            }
+        }
+    }}
 }
 
-/// List of useful RISC-V control and status registers
-#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
-pub enum RiscvCsr {
-    DPC,
-    PMPCFG0,
-    PMPCFG1,
-    PMPCFG2,
-    PMPCFG3,
-    PMPADDR0,
-    PMPADDR15,
+def_enum_name! {
+    /// List of useful RISC-V general purpose registers
+    #[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+    pub enum RiscvGpr {
+        RA,
+        SP,
+        GP,
+        TP,
+        T0,
+        T1,
+        T2,
+        FP,
+        S1,
+        A0,
+        A1,
+        A2,
+        A3,
+        A4,
+        A5,
+        A6,
+        A7,
+        S2,
+        S3,
+        S4,
+        S5,
+        S6,
+        S7,
+        S8,
+        S9,
+        S10,
+        S11,
+        T3,
+        T4,
+        T5,
+        T6,
+    }
+}
+
+def_enum_name! {
+    /// List of useful RISC-V control and status registers
+    #[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+    pub enum RiscvCsr {
```

Comment:
Will ibex's custom CSRs go in this enum too? Please could you `#[non_exhaustive]` if so?

Otherwise, would you mind updating the doc comments from "useful" to "all" for these two enums

Reply:
I removed "useful" for `RiscvGpr` and adds `#[non_exhaustive]` to `RiscvCsr`